src/qibo/backends/abstract.py

```diff
@@ -81,291 +143,227 @@ def set_dtype(self, dtype: str):  # pragma: no cover
-        """Generate :math:`|+++\\cdots+\\rangle` state vector as an array."""
-        raise_error(NotImplementedError)
+    @property
+    def int64(self):
+        return getattr(self.engine, "int64")
 
-    @abc.abstractmethod
-    def plus_density_matrix(self, nqubits: int):  # pragma: no cover
-        """Generate :math:`|+++\\cdots+\\rangle\\langle+++\\cdots+|` density matrix as an array."""
-        raise_error(NotImplementedError)
+    @property
+    def uint8(self):
+        return getattr(self.engine, "uint8")
 
-    @abc.abstractmethod
-    def matrix(self, gate: "qibo.gates.abstract.Gate"):  # pragma: no cover
-        """Convert a :class:`qibo.gates.Gate` to the corresponding matrix."""
-        raise_error(NotImplementedError)
+    ########################################################################################
+    ######## Methods related to array manipulation                                  ########
+    ########################################################################################
 
-    @abc.abstractmethod
-    def matrix_parametrized(self, gate: "qibo.gates.abstract.Gate"):  # pragma: no cover
-        """Equivalent to :meth:`qibo.backends.abstract.Backend.matrix` for parametrized gates."""
-        raise_error(NotImplementedError)
+    def abs(self, array, **kwargs) -> Union[int, float, complex, "ndarray"]:
+        return self.engine.abs(array, **kwargs)
 
-    @abc.abstractmethod
-    def matrix_fused(self, gate):  # pragma: no cover
-        """Fuse matrices of multiple gates."""
-        raise_error(NotImplementedError)
+    def all(self, array, **kwargs) -> Union["ndarray", bool]:
+        return self.engine.all(array, **kwargs)
 
-    @abc.abstractmethod
-    def apply_gate(self, gate, state, nqubits: int):  # pragma: no cover
-        """Apply a gate to state vector."""
-        raise_error(NotImplementedError)
+    def angle(self, array, **kwargs) -> "ndarray":
+        return self.engine.angle(array, **kwargs)
 
-    @abc.abstractmethod
-    def apply_gate_density_matrix(self, gate, state, nqubits: int):  # pragma: no cover
-        """Apply a gate to density matrix."""
-        raise_error(NotImplementedError)
+    def any(self, array, **kwargs) -> Union["ndarray", bool]:
+        return self.engine.any(array, **kwargs)
 
-    @abc.abstractmethod
-    def apply_gate_half_density_matrix(
-        self, gate, state, nqubits: int
-    ):  # pragma: no cover
-        """Apply a gate to one side of the density matrix."""
-        raise_error(NotImplementedError)
+    def arccos(self, array, **kwargs) -> "ndarray":
+        return self.engine.arccos(array, **kwargs)
 
-    @abc.abstractmethod
-    def apply_channel(self, channel, state, nqubits: int):  # pragma: no cover
-        """Apply a channel to state vector."""
-        raise_error(NotImplementedError)
+    def arctan2(self, array_1, array_2, **kwargs) -> "ndarray":
+        return self.engine.arctan2(array_1, array_2, **kwargs)
```

Comment:
Here and elsewhere, if you want to be more specific about the function signature that can be certainly something positive, since it is passing more information to the linter.

But then I would suggest to also specify the parameters' types:
```py
import numpy.typing as npt
```

```suggestion
    def arctan2(self, array_1: npt.ArrayLike, array_2: npt.ArrayLike, **kwargs) -> npt.ArrayLike:
        return self.engine.arctan2(array_1, array_2, **kwargs)
```

I'm not completely sure that `npt.ArrayLike` is the most suitable type. But it should be sufficient to encompass NumPy arrays, but also those from the other frameworks. Though I'm not completely sure, and we should double-check.

---

Alternatively, if you do not care, for the time being, about signatures, you could even compress the definition even more, with tuple unpacking
```suggestion
    def arctan2(self, *args, **kwargs) -> "ndarray":
        return self.engine.arctan2(*args, **kwargs)
```

---

All in all, even the present trade-off is a reasonable compromise. But let's assess the alternatives as well.